### ISSUE TITLE
Integration with Lightning Flash.

### ISSUE THREAD

**Issue by tchaton** [pystiche/pystiche#484]
Dear @pmeier,

This framework looks really neat !

I was wondering if you would be interested to collaborate to add a StyleTransferTask to Lightning Flash: https://github.com/PyTorchLightning/lightning-flash

Best,
T.C

**Comment by pmeier** [comment]
Hey @tchaton, sounds like a nice opportunity. Its just I have no experience with PyTorchLightning. Are you by any chance present at the PyTorch Ecosystem Day that is happening right now?

**Comment by pmeier** [comment]
@tchaton 

To re-iterate what we discussed at PTED21:

- Style transfer via model optimization should be added to `lightning-flash`
- We bench image optimization (at least for now) since it is a niche use case where no model is required
- `pystiche` can be added as dependency
- I will send a PR for this based on PyTorchLightning/lightning-flash#216

Did I miss anything?

**Comment by tchaton** [comment]
All good !

**Comment by pmeier** [comment]
@tchaton Unfortunately, this might take a while. I need to get a release out of the door to support newer PyTorch version, since I have a dependency clash with other libraries loaded by `lightning-flash`. I'll keep you posted.

**Comment by pmeier** [comment]
I've just finished `pystiche==0.7.1`. This should resolve the dependency clashes.

**Comment by pmeier** [comment]
Hey @tchaton. I encountered some problems where I have no idea how to solve them within `flash`:

- Models for Neural Style Transfer are trained in an unsupervised manner. Thus, I simply need a dataset that can supply images without labels / annotations.
- Following from the point above, there is no train / val / test split. The process is done after the training. If there is something like a validation / test it is performed manually by trying a few examples. There is no objective way to put a number on the quality of the stylization.
- The models used as transformer are not named. I've seen that the model is usually loaded by their name, which is thus not possible. We could fall back to a author / year combination of the paper the architecture was published.

**Comment by tchaton** [comment]
Hey @pmeier,

Mind opening a PR and joining Lightning Slack. Let s work this out together 🤗

Best,
T.C

**Comment by pmeier** [comment]
Closed in PyTorchLightning/lightning-flash#262.
